### DISCUSSION TITLE
How to get size of a directory?

### DISCUSSION THREAD

**Discussion by ArtjomP** [cake-build/cake#3390] [Q&A] [answered]
Hello there!

Cake has FileSize(filePath) for a file but it doesn't work for a directory.

Does Cake have an option to get the size of a directory?

Thank you in advance!

**Comment by augustoproiete** [comment, marked as answer]
Hey @ArtjomP, there's no equivalent alias for Directories, but you could easily use [`GetFiles`](https://cakebuild.net/dsl/globbing/#Files) to enumerate all the files in a directory and sum up the file sizes.

e.g.
```cakescript
var fileSizes = GetFiles("./**/*.*").Sum(FileSize);
Information("File sizes: {0}", fileSizes);
```

**Reply by ArtjomP** [reply]
Thank you!
It works well.
Even with the pattern: "./**/*"
Have a great time!
